### PR TITLE
implemented tag type filter

### PR DESCRIPTION
**Pull request by vincentcreusotfx**

### CODE REVIEW

**Review by rolandvarga — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -93,8 +93,10 @@ resource_groups:
 resource_tags:
   - resource_tag_name: "group"
     resource_tag_value: "tomonitor"
+    resource_types:
+     - "Microsoft.Compute/storage"
```
Comment (louisfelix) [inline README.md]: Why this extra space prefix (`resource_groups` example above do not have it)

File: README.md [hunk 2]
```diff
@@ -93,8 +93,10 @@ resource_groups:
...
     metrics:
-    - name: "CPU Credits Consumed"
+     - name: "CPU Credits Consumed"
```
Comment (louisfelix) [inline README.md]: Why this extra space prefix (`resource_groups` example above do not have it)

File: README.md [hunk 3]
```diff
@@ -127,6 +129,8 @@ Name of the tag to be filtered against.
 `resource_tag_value`:
 Value of the tag to be filtered against.
 
+`resource_types`: optional list of types kept in the list of resources gathered by tag. If the list is empty or the parameter is not defined, then all the resources are kept.
```
Comment (louisfelix) [inline README.md]:
```suggestion
`resource_types`: optional list of types kept in the list of resources gathered by tag. If the list is empty or the parameter is not defined, then all the resources are kept, and configured metric names must exist for every tagged resource.
```

File: azure.go [hunk 4]
```diff
@@ -248,7 +249,8 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
 }
 
 // Returns all resource with the given couple tagname, tagvalue
-func (ac *AzureClient) listByTag(tagName string, tagValue string) ([]string, error) {
+func (ac *AzureClient) listByTag(tagName string, tagValue string, types []string) ([]string, error) {
+	log.Println("Types ", types)
```
Comment (louisfelix) [inline azure.go]: Do you sill need this print?

File: azure.go [hunk 5]
```diff
@@ -286,7 +306,7 @@ func getAzureMonitorResponse(azureManagementEndpoint string) ([]byte, error) {
 		return nil, fmt.Errorf("Error creating HTTP request: %v", err)
 	}
 	req.Header.Set("Authorization", "Bearer "+ac.accessToken)
-
+	log.Printf("Request:%s", req.URL.String())
```
Comment (louisfelix) [inline azure.go]: Was this `Printf` kept on purpose?

File: config/config.go [hunk 6]
```diff
@@ -7,17 +7,17 @@ import (
 	"strings"
 	"sync"
 
-	yaml "gopkg.in/yaml.v2"
```
Comment (louisfelix) [inline config/config.go]: Last PR you did, Brian asked to leave that unchanged.

**Review by louisfelix — CHANGES_REQUESTED** [review]
Please add the tag example to `azure-example.yml`

File: azure.go (hunk 4 above)
Comment (vincentcreusotfx) [reply azure.go]: Nope !

File: azure.go (hunk 5 above)
Comment (vincentcreusotfx) [reply azure.go]: On purpose no good catch

File: config/config.go (hunk 6 above)
Comment (vincentcreusotfx) [reply config/config.go]: yeah nice catch too

File: README.md (hunk 1 above)
Comment (vincentcreusotfx) [reply README.md]: My formatter did add that. I tested both to be sure, both are working so I'm applying the already existing way.

File: README.md (hunk 2 above)
Comment (vincentcreusotfx) [reply README.md]: Same as above

**Review by louisfelix — APPROVED** [review]

**Pull request merged**